p5/sketch/Skia2DRenderer/renderer2d.py

```diff
@@ -109,30 +163,55 @@ def render_image(self, img, *args):
         self.canvas.drawImage(img, *args)
 
     def clear(self):
-        self.canvas.clear(skia.Color4f(*self.style.background_color))
+        # Our screen will go black or all the pixels would be transparent
+        self.canvas.clear(skia.Color(0, 0, 0, 0))
+
+    def background(self, *args, **kwargs):
+        # TODO: Add if args == pImage check
+        # TODO: Add blend mode logic later
+        self.push_matrix()
+        self.reset_matrix()
+
+        curr_fill = self.style.fill_color
+        curr_fill_enabled = self.style.fill_enabled
+        curr_stroke_enabled = self.style.stroke_enabled
+
+        self.style.fill_enabled = True
+        self.style.stroke_enabled = False
+        self.style.fill_color = Color(*args, **kwargs).normalized
+        self.rect(0, 0, *p5.sketch.size)
+        self.style.fill_color = curr_fill
+        self.style.fill_enabled = curr_fill_enabled
+        self.style.stroke_enabled = curr_stroke_enabled
 
-    def render(self, fill=True, stroke=True):
+        self.pop_matrix()
+
+    def render(self, fill=True, stroke=True, rewind=True):
         """
         Draw the path on current canvas using paint
         """
         # TODO: Check, do we really have to setColor, and setStyle on each render call
         # TODO: Explore ways to do this, such that it works with pGraphics as well
         if self.style.fill_enabled and fill:
+            self.paint.setStrokeCap(self.style.stroke_cap)
+            self.paint.setStrokeJoin(self.style.stroke_join)
```

Comment:
Do we need to set `StrokeCap` and `StrokJoin` in both `if self.style.fill_enabled` and `if self.style.stroke_enabled`?

Reply:
Nice catch, yes it should only be set and used in `stroke_enabled` if clause.